app/controllers/concerns/password_helpers.rb

```diff
@@ -20,11 +20,10 @@ def update_password
 
   def deliver_reset_password_instructions(user)
     token, enc = Devise.token_generator.generate(user.class, :reset_password_token)
-    user.reset_password_token   = enc
+    user.reset_password_token = enc
     user.reset_password_sent_at = Time.now.utc
     user.save(validate: false)
-    message = DeviseMailer.reset_password_instructions(user, token, current_user.name)
-    message.deliver_later
+    DeviseMailer.reset_password_instructions(user, token, current_user.name)
```

Comment:
Not sure why we are no longer using deliver_later here?

Reply:
because it has no effect and is actually misleading. Devise ends up overriding the `deliver_later` and actually doing a `deliver_now`, as mentioned in the commit message - these messages ended up not running in the worker/via-sidekiq as a result (for years)

Reply:
This commit is actually atomic and could go in on its own to correct the existing bug

Reply:
👍

Reply:
I think i will add a comment re devise here actually as it need to be clear to other devs